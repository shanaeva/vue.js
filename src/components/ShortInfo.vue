<template>
  <div
    v-for="info in infoList"
    :key="info.description"
    class="short-info"
  >
    <p class="value">
      {{ info.value }}
    </p>
    <p class="description">
      {{ info.description }}
    </p>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue';

type InfoType = {
  value: number,
  description: string,
};

export default defineComponent({
  name: 'ShortInfo',
  props: {
    infoList: { type: Array as PropType<Array<InfoType>>, required: true },
  },
});
</script>

<style scoped>
  .short-info{
    display: flex;
    align-items: flex-end;
    margin-right: 24px;
  }
  .value {
    margin-right: 6px;
    color: #f65261;
    font-size: 28px;
  }
  .description {
    color: #555555;
  }
</style>
