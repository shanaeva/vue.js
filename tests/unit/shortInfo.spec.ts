import { shallowMount } from '@vue/test-utils';
import ShortInfo from '@/components/ShortInfo.vue';

describe('ShortInfo', () => {
  it('render', () => {
    const wrapper = shallowMount(ShortInfo, { props: { infoList: [{ value: 1, description: 'year' }] } });
    expect(wrapper.props().infoList).toStrictEqual([{ value: 1, description: 'year' }]);
  });
  it('render correctly', () => {
    const wrapper = shallowMount(ShortInfo, { props: { infoList: [{ value: 1, description: 'year' }] } });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
