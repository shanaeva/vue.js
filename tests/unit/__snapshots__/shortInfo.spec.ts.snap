// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortInfo render correctly 1`] = ``;
